### PR TITLE
CDAP-20841: Add stop operation api

### PR DESCRIPTION
**Pull request by samdgupi**
Add api to send stop operation message

### CODE REVIEW

**Review by GnsP — APPROVED** [review]

**Comment by GnsP** [comment]
There is one checkstyle issue regarding the length of a line. Please fix that before merging.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java]: this sort of business logic should be in the lifecycle manager instead of in the handler. In general, http handlers should not be doing much more than deserializing user input and translating internal objects and errors into responses.

**Review by albertshau — COMMENTED** [review]
small comment

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java
Comment (chtyim) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java]: Just curious, where do we pick this message and perform the actual stopping of the running task?

Comment (samdgupi) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java]: We consume the message in https://github.com/cdapio/cdap/blob/develop/cdap-app-fabric/src/main/java/io/cdap/cdap/internal/operation/OperationNotificationSingleTopicSubscriberService.java#L191

File: cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java
Comment (samdgupi) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java]: Move the logic to lifecycle manager

**Comment by samdgupi** [comment]
@albertshau Incorporated the suggestion. Merging the pull request to unblock e2e testing for 6.10. In case any further suggestions will take care in a followup PR

**Pull request merged**
